.github/workflows/upload.yml

```diff
@@ -0,0 +1,36 @@
+name: Upload lte-core-interface-lib
+
+on:
+  push:
+    branches:
+      - "main"
+
+jobs:
+  charmhub-upload:
+    runs-on: ubuntu-20.04
+    name: Charmhub upload lib
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Wait for CI to succeed
+        uses: lewagon/wait-on-check-action@v1.1.2
+        with:
+          ref: main
+          check-name: "CI"
```

Comment:
`check-name` is the name of the job not the name of the workflow. 
In this case, we probably want to depend on `Unit tests`.

Reply:
Fixed.